src/Apropos/Gen.hs

```diff
@@ -154,13 +191,16 @@ scale s g = liftF (Scale s g id)
 prune :: Gen a -> Gen a
 prune g = liftF (Prune g id)
 
+liftPropertyT :: PropertyT IO a -> Gen a
+liftPropertyT p = liftF (LiftPropertyT p id)
```

Comment:
Is this too much power? It allows users to perform arbitrary IO in Gen...

We could have `liftTraversal :: forall p m . Traversal p m -> Gen m`.

Reply:
This is nicer I think. Change these type sigs from Has[Parameterised/Permutation]Generator back to these

`parameterisedGenerator :: Set p -> Gen m`
`buildGen :: Gen m -> Set p -> Gen m`

and make the `Traversal` internal.